### PR TITLE
[MER-511] Horizontal scaling

### PR DESCRIPTION
**Pull request by mcass19**
https://eliterate.atlassian.net/browse/MER-511

- We use [libcluster](https://github.com/bitwalker/libcluster) for automatic cluster formation
- We are using the `Gossip` strategy for default config
- For prod env (releases), we will use the [EC2 strategy](https://github.com/kyleaa/libcluster_ec2) that uses `ex_aws` to query the EC2 DescribeInstances API endpoint. Access to this API should be granted to the EC2 instance profile.
- Take into account that different strategies could use different config options

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -189,6 +189,15 @@ config :oli, :footer,
   link_2_location: System.get_env("FOOTER_LINK_2_LOCATION", ""),
   link_2_text: System.get_env("FOOTER_LINK_2_TEXT", "")
 
+# Configure libcluster for horizontal scaling
+# Take into account that different strategies could use different config options
+config :libcluster,
+  topologies: [
+    oli: [
+      strategy: Module.concat([System.get_env("LIBCLUSTER_STRATEGY", "Cluster.Strategy.Gossip")])
```
Comment (gastonabella) [inline config/config.exs:197]: Is it necessary to use `Module.concat` here? We are passing a one-element-list and I think it's used when you want to concat several aliases.

File: config/releases.exs [hunk 2]
```diff
@@ -198,6 +198,20 @@ config :oli, :footer,
   link_2_location: System.get_env("FOOTER_LINK_2_LOCATION", ""),
   link_2_text: System.get_env("FOOTER_LINK_2_TEXT", "")
 
+# Configure libcluster for horizontal scaling
+# Take into account that different strategies could use different config options
+config :libcluster,
+  topologies: [
+    oli: [
+      strategy:
+        Module.concat([System.get_env("LIBCLUSTER_STRATEGY", "ClusterEC2.Strategy.Tags")]),
```
Comment (gastonabella) [inline config/releases.exs:207]: Same here

**Review by gastonabella — APPROVED** [review]
Awesome job! 👏 🚀 

I only left a minor comment

File: config/config.exs (hunk 1 above)
Comment (mcass19) [reply config/config.exs:197]: I totally agree but I didn't find a quick way to build a module name from a string, do you have some suggestion?

**Review by jrissler — APPROVED** [review]
Looks great - love how simple it is to enable this.

**Pull request closed without merging**